src/runtime_src/core/pcie/driver/linux/xocl/subdev/xvc.c

```diff
@@ -170,8 +171,19 @@ static long xvc_ioctl_helper(struct xocl_xvc *xvc, const void __user *arg)
 	}
 
 	total_bits = xvc_obj.length;
+
+	/* Fixing integer overflow scenario */
+	if (total_bits >= UINT_MAX - 7)
+		total_bits = UINT_MAX - 7;
+
 	total_bytes = (total_bits + 7) >> 3;
 
+	if (total_bytes == 0) {
+		pr_info("Err: int overflow, op 0x%x, len %u bits, %u bytes.\n",
```

Comment:
is this error valid if total_bits is zero? would there be a case where total_bits can be zero?

Reply:
Not sure if driver receives total_bits as 0. But, If total_bits is 0, so total_bytes will also be 0. In the next step, driver tries to allocate mem for 0 bytes, which might be an issue. So, added a check for total_bits.